aicsimageio/tests/test_transforms.py

```diff
@@ -30,6 +30,64 @@
         (da.zeros((6, 200, 400)), "ZYX", "STCZYX", {}, (1, 1, 1, 6, 200, 400)),
         (np.zeros((6, 200, 400)), "ZYX", "ZCYSXT", {}, (6, 1, 200, 1, 400, 1)),
         (da.zeros((6, 200, 400)), "ZYX", "ZCYSXT", {}, (6, 1, 200, 1, 400, 1)),
+        (np.zeros((6, 200, 400)), "ZYX", "CYSXT", {"Z": 2}, (1, 200, 1, 400, 1)),
```

Comment:
If every test in these cases has both a `da.zeros` and `np.zeros` equivalent, I might recommend adding an additional parametrize which is `array_maker` (or something similar) that is either of the two functions. `data` would just become a list of tuples that then have `array_maker` applied on them as the first operation of the test. This would reduce the number of parameterized cases you need to make/repeat but keep the final number of tests.

Reply:
so something like:
```python
@pytest.mark.parametrize("data", [
    np.zeros((6, 200, 400)),
    da.zeros((6, 200, 400)),
])
@pytest.mark.parametrize("the, rest, of, the, kwargs", [
    ...
])
def test_function(data, the, rest, of, the, kwargs)
```

Reply:
Hmmmm this isn't immediately possible given that the shapes of images change every so often and the other params are dependent on shape...

Reply:
Here's my thought - currently you have this:
```python
@pytest.mark.parametrize("data, the, rest, of, the, kwargs", [
    (np.zeros((6, 200, 400)), ...),
    (da.zeros((6, 200, 400)), ...),
    # everything repeated twice for np.zeros & da.zeros
])
def test_function(data, the, rest, of, the, kwargs)
```
I might suggest the following:
```python
@pytest.mark.parametrize("array_maker", [np.zeros, da.zeros])
@pytest.mark.parametrize("data_shape, the, rest, of, the, kwargs", [
    ((6, 200, 400), ...),
    # Only one line for the test case
])
def test_function(array_maker, data_shape, the, rest, of, the, kwargs):
    data = array_maker(data_shape)
    # Rest of the test goes unchanged
```
This would reduce the duplicates you have since you only have to specify the shape once, and parametrize would handle the Cartesian-product of applying both `array_maker`s to each other test case!

Reply:
You're a genius